WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Avalonia.Controls;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.ViewModels.CoinControl.Core;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Fluent.ViewModels.Wallets.Send;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinControl;
+
+[NavigationMetaData(
+	Title = "Select Coins",
+	Caption = "",
+	IconName = "wallet_action_send",
+	NavBarPosition = NavBarPosition.None,
+	Searchable = false,
+	NavigationTarget = NavigationTarget.DialogScreen)]
+public partial class SelectCoinsDialogViewModel : DialogViewModelBase<IEnumerable<SmartCoin>>
+{
+	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel, TransactionInfo info, IEnumerable<SmartCoin> transactionSpentCoins)
+	{
+		var pocket = walletViewModel.Wallet.GetPockets();
+
+		var nodes = ToNodes(pocket);
+
+		Source = new HierarchicalTreeDataGridSource<TreeNode>(nodes)
+		{
+			Columns =
+			{
+				ColumnFactory.ChildrenColumn(),
+				ColumnFactory.IndicatorsColumn(),
+				ColumnFactory.AmountColumn(),
+				ColumnFactory.PrivacyScore(),
+				ColumnFactory.PocketColumn()
+			}
+		};
+
+		Source.SortBy(Source.Columns[4], ListSortDirection.Descending);
+
+		SetupCancel(false, true, false);
+		EnableBack = true;
+	}
+
+	public HierarchicalTreeDataGridSource<TreeNode> Source { get; }
+
+	private static IEnumerable<TreeNode> ToNodes(IEnumerable<Pocket> pockets)
```

Comment:
Do we need separate static method 
```
		var nodes = walletViewModel.Wallet
			.GetPockets()
			.Select(p => new TreeNode(new PocketCoinAdapter(p), p.Coins.OrderByDescending(x => x.Amount).Select(coin => new TreeNode(new SmartCoinAdapter(coin)))));
		```

Reply:
I think this:

```
var pockets = walletViewModel.Wallet.GetPockets();
var nodes = ToTreeNodes(pockets);
```

Reads better than this:

```
var pockets = walletViewModel.Wallet.GetPockets();

var nodes = pockets.Select(
	pocket => new TreeNode(
		new PocketCoinAdapter(pocket),
		pocket.Coins
			.OrderByDescending(x => x.Amount)
			.Select(coin => new TreeNode(new SmartCoinAdapter(coin)))));
```

The method is intention-revealing, while the raw code is a bit more obscure without the proper name.